Tools/ci-run.sh

```diff
@@ -3,125 +3,164 @@
+elif [[ $PYTHON_VERSION == "3."[45]* ]]; then
   python -m pip install wheel || exit 1
   python -m pip install -r test-requirements-34.txt || exit 1
 else
   python -m pip install -U pip setuptools wheel || exit 1
 
-  if [ -n "${PYTHON_VERSION##*-dev}" ]; then
+  if [[ $PYTHON_VERSION == *"-dev" || $COVERAGE == "1" ]]; then
     python -m pip install -r test-requirements.txt || exit 1
 
-    if [ "${PYTHON_VERSION##pypy*}" -a "${PYTHON_VERSION##3.[4789]*}" ]; then
+    if [[ $PYTHON_VERSION == "pypy"* && $PYTHON_VERSION == "3."[4789]* ]]; then
       python -m pip install -r test-requirements-cpython.txt || exit 1
     fi
   fi
 fi
 
-if [ "$TEST_CODE_STYLE" == "1" ]; then
-  STYLE_ARGS="--no-unit --no-doctest --no-file --no-pyregr --no-examples";
+if [[ $TEST_CODE_STYLE == "1" ]]; then
+  STYLE_ARGS="--no-unit --no-doctest --no-file --no-pyregr --no-examples"
   python -m pip install -r doc-requirements.txt || exit 1
 else
-  STYLE_ARGS="--no-code-style";
+  STYLE_ARGS="--no-code-style"
 
   # Install more requirements
-  if [ -n "${PYTHON_VERSION##*-dev}" ]; then
-    if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $PYTHON_VERSION == *"-dev" ]]; then
+    if [[ $BACKEND == *"cpp"* ]]; then
       echo "WARNING: Currently not installing pythran due to compatibility issues"
       # python -m pip install pythran==0.9.5 || exit 1
-    fi
-
-    if [ "$BACKEND" != "cpp" -a -n "${PYTHON_VERSION##pypy*}" -a -n "${PYTHON_VERSION##2*}" -a -n "${PYTHON_VERSION##*3.4}" ]; then
+    elif [[ $PYTHON_VERSION == "pypy"* && $PYTHON_VERSION == "2"* && $PYTHON_VERSION == *"3.4" ]]; then
       python -m pip install mypy || exit 1
     fi
   fi
 fi
 
 # Run tests
+echo "==== Running tests ===="
 ccache -s 2>/dev/null || true
 export PATH="/usr/lib/ccache:$PATH"
 
-if [ "$NO_CYTHON_COMPILE" != "1" -a -n "${PYTHON_VERSION##pypy*}" ]; then
-  CFLAGS="-O2 -ggdb -Wall -Wextra $(python -c 'import sys; print("-fno-strict-aliasing" if sys.version_info[0] == 2 else "")')" \
-  python setup.py build_ext -i \
-          $(if [ "$COVERAGE" == "1" ]; then echo " --cython-coverage"; fi) \
-          $(if [ "$CYTHON_COMPILE_ALL" == "1" ]; then echo " --cython-compile-all"; fi) \
-          $(python -c 'import sys; print("-j5" if sys.version_info >= (3,5) else "")') \
-      || exit 1
-  if [ -z "$COVERAGE" -a -z "$STACKLESS" -a -z "$LIMITED_API" -a -z "$CYTHON_COMPILE_ALL" -a -z "$EXTRA_CFLAGS" -a -n "${BACKEND//*cpp*}" ]; then
+if [[ "$OSTYPE" == "msys" ]]; then  # for MSVC cl
+  # /wd will disable warning
+  # 4711 warns that function `x` selected for automatic inline expansion
+  # 4127 warns that conditional expression is constant, should be fixed here https://github.com/cython/cython/pull/4317
+  # (off by default) 5045 warns that Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
+  # (off by default) 4820 warns about the code in Python\3.9.6\x64\include ...
+  WARNARGS="/W4 /wd4711 /wd4127 /wd5045 /wd4820"
```

Comment:
`\W4` is recommended and gives an adequate number of warnings, but we can use ` \Wall` + ignore unnecessary warnings (additionally to those `/wd`s)